src/caffeine/prif_private_s.f90

```diff
@@ -3,7 +3,7 @@
 submodule(prif) prif_private_s
   implicit none
 
-  type(prif_team_type), target :: default_team
+  type(prif_team_type), target :: initial_team
   type(prif_team_type), pointer :: current_team => null()
```

Comment:
Now that `prif_team_type` has been corrected to act as a pointer, we no longer really need the `pointer` and `target` attributes above. 

`current_team` can simply be a (non-pointer) `type(prif_team_type)` with `current_team%info` pointing to the appropriate `team_data`. Similarly for `initial_team`.

```suggestion
  type(prif_team_type) :: initial_team
  type(prif_team_type) :: current_team
```

This will also require a 1-line change in `prif_init` where we initialize `current_team`.

Reply:
As a further improvement, we might eventually want to statically allocate the initial team data, ie:

 ```suggestion
  type(team_data), target :: initial_team
  type(prif_team_type) :: current_team 
```

The initial team is "special" for many reasons, especially because so many calls in PRIF now accept an image index that is relative to the initial team (ignoring the current team). This change would allow Caffeine to skip a pointer indirection when doing things like sanity checking that the client provided a valid image index in the initial team.